Dockerfile

```diff
@@ -1,17 +1,38 @@
-FROM node:10 as builder
-WORKDIR /app
-COPY package*.json ./
-RUN npm install
-COPY . .
+FROM node:12-slim as base
+RUN mkdir /app && chown -R node:node /app
+USER node
 ARG APP_MOUNT_URI
 ARG API_URI
 ARG STATIC_URL
 ENV API_URI ${API_URI:-http://localhost:8000/graphql/}
-ENV APP_MOUNT_URI ${APP_MOUNT_URI:-/dashboard/}
-ENV STATIC_URL ${STATIC_URL:-/dashboard/}
-RUN STATIC_URL=${STATIC_URL} API_URI=${API_URI} APP_MOUNT_URI=${APP_MOUNT_URI} npm run build
-
-FROM nginx:stable
+ENV APP_MOUNT_URI ${APP_MOUNT_URI:-/}
+ENV STATIC_URL ${STATIC_URL:-/}
+ENV PATH=/app/node_modules/.bin:$PATH
 WORKDIR /app
-COPY ./nginx/default.conf /etc/nginx/conf.d/default.conf
-COPY --from=builder /app/build/ /app/
+COPY --chown=node:node package*.json ./
+RUN npm install
+
+FROM base as dev
+CMD npm start
+
+FROM base as source
+COPY --chown=node:node assets ./assets/
+COPY --chown=node:node locale ./locale/
+COPY --chown=node:node testUtils ./testUtils/
+COPY --chown=node:node src ./src/
+COPY --chown=node:node *.js ./
+COPY --chown=node:node *.json ./
+COPY --chown=node:node *.ts ./
```

Comment:
Rather than having one layer per type of files, you could only create one:
```dockerfile
COPY --chown=uid:gid \
        assets \
        locale \
        testUtils \
        etc. \
        /app/
```

Reply:
Didn't know that, fixed